### PR TITLE
MRG: Add raw PSD to parse_folder

### LINKED ISSUE #5237: Add spectra to report function?
Maybe it would make sense to add spectra to the reports. They can be very informative.

Another idea would be epochs images, which can also be highly diagnostic.

Not volunteering :)

@jasmainak

### PR DESCRIPTION
**Pull request by larsoner**
Closes #5237.

Also shifts our docs from `mne.report` namespace to the root namespace as `mne.Report` already lives there.

Copying the truncated testing raw file to desktop and doing:
```
import mne
report = mne.Report(raw_psd=True)
report.parse_folder('/Users/larsoner/Desktop')
report.save('/Users/larsoner/Desktop/report.html')
```
I get:

screen shot 2018-08-04 at 17 52 48

@jona-sassenhagen happy?

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
Looks nice already! I'll take a closer look tomorrow!

File: examples/visualization/make_report.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 # Do standard folder parsing (this can take a couple of minutes):
 
 report = Report(image_format='png', subjects_dir=subjects_dir,
-                info_fname=evoked_fname, subject='sample')
+                info_fname=evoked_fname, subject='sample', raw_psd=True)
```
Comment (jasmainak) [inline examples/visualization/make_report.py]: why is it a boolean flag? why not plot it always?

File: mne/report.py [hunk 2]
```diff
@@ -314,26 +312,31 @@ def _update_html(html, report_fname, report_sectionlabel):
...
         figsize = tuple(figsize[1:])
         data = data[:, :, 0]
-    fig = Figure(figsize=figsize, dpi=1.0, frameon=False)
-    FigureCanvas(fig)
+    fig = _figure_agg(figsize=figsize, dpi=92, frameon=False)
     cmap = getattr(plt.cm, cmap, plt.cm.gray)
     fig.figimage(data, cmap=cmap)
     output = BytesIO()
-    fig.savefig(output, dpi=1.0, format='png')
+    fig.savefig(output, dpi=92, format='png')
```
Comment (jasmainak) [inline mne/report.py]: why did you change this?

File: mne/report.py [hunk 3]
```diff
@@ -1582,6 +1596,19 @@ def _render_raw(self, raw_fname):
...
             meas_date=meas_date, n_eeg=n_eeg, n_grad=n_grad, n_mag=n_mag,
             eog=eog, ecg=ecg, tmin=tmin, tmax=tmax)
+
+        if isinstance(self._raw_psd, dict):
+            from matplotlib.backends.backend_agg import FigureCanvasAgg
+            n_ax = sum(kind in raw for kind in _data_types)
+            fig, axes = plt.subplots(n_ax, 1, figsize=(6, 1 + 1.5 * n_ax),
+                                     dpi=92)
+            FigureCanvasAgg(fig)
+            img = _fig_to_img(raw.plot_psd, self.image_format,
```
Comment (jasmainak) [inline mne/report.py]: you compute it on the entire data?

File: mne/report.py [hunk 4]
```diff
@@ -1582,6 +1596,19 @@ def _render_raw(self, raw_fname):
             div_klass='raw', id=global_id, caption=caption, info=raw.info,
             meas_date=meas_date, n_eeg=n_eeg, n_grad=n_grad, n_mag=n_mag,
             eog=eog, ecg=ecg, tmin=tmin, tmax=tmax)
+
+        if isinstance(self._raw_psd, dict):
+            from matplotlib.backends.backend_agg import FigureCanvasAgg
+            n_ax = sum(kind in raw for kind in _data_types)
+            fig, axes = plt.subplots(n_ax, 1, figsize=(6, 1 + 1.5 * n_ax),
+                                     dpi=92)
+            FigureCanvasAgg(fig)
```
Comment (jasmainak) [inline mne/report.py]: why not `_figure_agg` here?

File: mne/report.py (hunk 3 above)
Comment (jona-sassenhagen) [reply mne/report.py]: I think it would actually be wisest to compute it on the whole data -- for example if there is a huge artefact at the end of the data, only checking the first minute would not catch it. Can't it be sped up in other ways?

Actually, computing psds never takes too long for me?

File: mne/report.py (hunk 4 above)
Comment (larsoner) [reply mne/report.py]: Because it won't work with `plt.subplots` call

File: mne/report.py (hunk 3 above)
Comment (larsoner) [reply mne/report.py]: > you compute it on the entire data?

By default it does what `raw.plot_psd` will do, and yes we changed the default to be the entire data a while ago. This is implicitly controllable via the `dict` option.

> Actually, computing psds never takes too long for me?

It can take 30+ sec on some files. You don't have enough channels and/or time points :)

File: mne/report.py (hunk 2 above)
Comment (larsoner) [reply mne/report.py]: 1.0 was a strange choice and seemed like a bug

File: examples/visualization/make_report.py (hunk 1 above)
Comment (larsoner) [reply examples/visualization/make_report.py]: Backward compat / not changing the output. I didn't want to force a slow-to-compute option on people, but if it seems useful enough I guess it's easy to disable so we could switch it to `True` default.

File: mne/report.py (hunk 3 above)
Comment (jona-sassenhagen) [reply mne/report.py]: I don’t think 30s is long i guess.

File: mne/report.py (hunk 2 above)
Comment (agramfort) [reply mne/report.py]: it does not expand the size of a regular report? these can easily become huge. Maybe it was the tradoff?

File: mne/report.py [hunk 5]
```diff
@@ -840,7 +851,11 @@ def __init__(self, info_fname=None, subjects_dir=None,
         # boolean to specify if sections should be ordered in natural
         # order of processing (raw -> events ... -> inverse)
         self._sort_sections = False
-
+        raw_psd = {} if raw_psd is True else raw_psd
+        if raw_psd is not False and not isinstance(raw_psd, dict):
+            raise TypeError('raw_psd must be bool or dict, got %s'
+                            % (type(raw_psd),))
+        self._raw_psd = raw_psd
```
Comment (agramfort) [inline mne/report.py]: following the sklearn way to go I would just do

self. raw_psd = raw_psd

after checking it's a bool or dict and do the conversion to dict in the _render_raw method without adding another attribute to the object.

File: mne/report.py [hunk 6]
```diff
@@ -1290,8 +1305,16 @@ class construction.
...
+            # only warn if relevant
+            if any(fname.endswith(('-cov.fif', '-cov.fif.gz'))
+                   for fname in fnames):
+                warn('`info_fname` not provided. Cannot render '
+                     '-cov.fif(.gz) files.')
+            if any(fname.endswith(('-trans.fif', '-trans.fif.gz'))
+                   for fname in fnames):
+                warn('`info_fname` not provided. Cannot render '
+                     '-trans.fif(.gz) files.')
+                raise RuntimeError
```
Comment (agramfort) [inline mne/report.py]: now it's an error while it was just a warning. Can it risk breaking code that was running before?

**Comment by agramfort** [comment]
besides my nitpickS LGTM

File: mne/report.py (hunk 6 above)
Comment (larsoner) [reply mne/report.py]: oof yes this is just debugging cruft...

**Comment by larsoner** [comment]
Comments addressed. @jasmainak do you have time to look again?

**Comment by agramfort** [comment]
@jasmainak merge if you're happy

**Comment by jasmainak** [comment]
Do we also want to plot the PSD from the command line? If so, it should be exposed in the command line arguments.

**Comment by larsoner** [comment]
I'm not sure it's worth exposing every param via command line. Especially this one since people may well want to use non-default args (like start and end values)...

**Comment by jasmainak** [comment]
@larsoner what did you decide about the `dpi` settings? If I remember correctly, it was @dengemann who wrote that part of the report

**Comment by jasmainak** [comment]
Other than these two comments, it's good to go from my end.

**Comment by larsoner** [comment]
I reverted the DPI changes you commented on

**Comment by jasmainak** [comment]
okay, thanks @larsoner for the PR and the clarifications !
